src/context.jl

```diff
@@ -0,0 +1,169 @@
+using IRTools: isexpr, IR, @dynamo, postwalk
```

Comment:
maybe name this file (and wherever the term is used) `contextual.jl` re CUDAnative.jl's Cassette PR. CUDA already has context objects, so that might be confusing.